src/Collection.php

```diff
@@ -967,17 +872,13 @@ public function mapReduce(JavascriptInterface $map, JavascriptInterface $reduce,
             $options['readConcern'] = $this->readConcern;
         }
 
-        if (! isset($options['typeMap'])) {
-            $options['typeMap'] = $this->typeMap;
-        }
-
-        if (! isset($options['writeConcern']) && ! is_in_transaction($options)) {
-            $options['writeConcern'] = $this->writeConcern;
-        }
+        $options = $this->inheritWriteOptions(
+            $this->inheritTypeMap($options),
```

Comment:
Thinking about how you had to retain `if (! isset($options['readPreference']) && ! is_in_transaction($options))` in this method, did you consider splitting the ReadPreference and ReadConcern inheritance into their own methods (like TypeMap and Codec) and then having a common `inheritReadOptions()` call out to the individual methods?

The special handling for ReadConcern inheritance only comes up a small number of times so I'm fine either way. Just curious if this was intentional.

Reply:
I went back and added `inheritReadConcern` and `inheritReadPreference`, simplifying the conditions above as well.